### DISCUSSION TITLE
NATS Terminology Clarification

### DISCUSSION THREAD

**Discussion by gcolliso** [nats-io/nats.docs#365] [Ideas]
We will use this discussion to list NATS terminology that may need additional clarification and thus need additional documentation.

**Comment by nav9** [comment]
Thank you for this. Starting off:
  
These points are from the [Request reply page](https://docs.nats.io/nats-concepts/core-nats/reqreply).
1. It would help to understand more about “_drain before exiting_”, to explain it to newcomers. How do we explain this? What kind of resources does NATS drain from memory before exiting?  
2. Similarly, “_NATS allows multiple responders to run and form dynamic queue groups for transparent scale up_” needs simpler language. Could we say "_Multiple NATS responders can form dynamic queue groups. This allows the responders to scale up as per demand._ ". What does the "dynamic" mean in "dynamic queue groups? Does each publisher and subscriber constitute a group? So when a subscriber replies to a publisher, is that message treated as being part of a queue? Or does the publisher maintain multiple queues to cater to the responses from each subscriber?  
3. One would also need to know what "_observability_" is in “_observability is as simple as running..._”.   
4. When mentioning “_Try NATS request-reply on your own, using a live server_”, one would want to have a link to the live server (if the NATS team intends to make it available), or the user needs to be pointed to a page which helps them setup their own live server (which page would you recommend?). 
5.  The diagram on the page does not  capture all aspects  of what is explained. We need to show the queue groups and show some way in which the scaling up  happens. Any suggestions on what you'd like to present as part of the diagram?

**Reply by timove** [reply]
IIRC it was confusing for me at first that there are streams in Jetstream as well as in export/import.
